Submitting for review

Kamalesh-Radhakrishnan opened the pull request.

Kamalesh-Radhakrishnan committed:
Submitting for review

ramkrishk commented on kids-learning-app/assets/css/styles.css:
```diff
@@ -0,0 +1,178 @@
+    height: 50px;
+}
+
+.col1{
+    width: 200px;
+    margin: 0;
+    padding-top: 15px;
+    padding-left: 10px;
+    background-color: #2b60ce;
+    color: #ffffff;
+    text-align: left;
+    font-size: 15px;
+}
+
+.col2{
+    width:150px;
+    margin: 0;
+    padding-top: 15px;
+    padding-left: 10px;
+    background-color: #4a5d7c;
+    color: #ffffff;
+    text-align: left;
+    font-size: 15px;
+}
+
+.col3{
+    width:150px;
+    margin: 0;
+    padding-top: 15px;
+    padding-left: 10px;
+    background-color: rgb(75,134,247);
+    color: #ffffff;
+    text-align: left;
+    font-size: 15px;
+}
```
There are few redundant CSS properties with the same value, think of a solution that can prevent this

ramkrishk commented on kids-learning-app/assets/js/app.js:
```diff
@@ -0,0 +1,260 @@
+function calculateFunc(){
+    var lsobj = JSON.parse(localStorage.getItem("lsShape"));
+    var dimension = parseInt(lsobj.dimension);
+    if(lsobj.shape == 'circle'){
+        lsobj.perimeter = (2 * Math.PI * dimension).toFixed(2) + " cm";
+        lsobj.area = (Math.PI * dimension * dimension).toFixed(2) + " sq cm";
+        lsobj.dimensionName = "RADIUS";
+        lsobj.dimensionFormula = "r";
+        lsobj.perimeterFormula = "2 * 𝝅 * r";
+        lsobj.areaFormula = "𝝅 * r * r";
+    }else if(lsobj.shape == 'triangle'){
+        lsobj.perimeter = (3 * dimension).toFixed(2) + " cm";
+        lsobj.area = (0.433 * dimension * dimension).toFixed(2) + " sq cm";
+        lsobj.dimensionName = "SIDE";
+        lsobj.dimensionFormula = "s";
+        lsobj.perimeterFormula = "3 * s";
+        lsobj.areaFormula = "0.433 * s * s";
+    }else if(lsobj.shape == 'square'){
+        lsobj.perimeter = (4 * dimension).toFixed(2) + " cm";
+        lsobj.area = (dimension * dimension).toFixed(2) + " sq cm";
+        lsobj.dimensionName = "SIDE";
+        lsobj.dimensionFormula = "a";
+        lsobj.perimeterFormula = "4 * a";
+        lsobj.areaFormula = "a * a";
+    }
```
What next? think of this problem in an OOPs way.

ramkrishk commented on kids-learning-app/assets/js/app.js:
```diff
@@ -0,0 +1,260 @@
+        var titleText = components.titleDiv();
+        var shape="";
+
+        switch(lsobj.shape){
+            case('circle'):
+                shape = components.circleButton();
+                titleText.innerText = "Circle";
+                break;
+            case('triangle'):
+                shape = components.triangleButton();
+                titleText.innerText = "Equilateral Triangle";
+                break;
+            case('square'):
+                shape = components.squareButton();
+                titleText.innerText = "Square";
+                break;
+        };
```
To add a new shape we'd need to add a switch case again, think of a solution

ramkrishk replied:
Same comment applies for the above switch statement

ramkrishk commented on kids-learning-app/assets/js/app.js:
```diff
@@ -0,0 +1,260 @@
+        shapes.append(circle,triangle,square);
+        body.append(title,shapes);
+
+        document.querySelectorAll(".fa-check").forEach(el => el.style.display = "none");
```
rather inline style, add or remove class that'd manipulates the properties

ramkrishk replied:
make changes in other places

ramkrishk commented on kids-learning-app/assets/js/app.js:
```diff
@@ -0,0 +1,260 @@
+        return chkicon;
+    },
+
+    circleButton : () => {
+        let circle = document.createElement('div');
+        circle.setAttribute("class","circle");
+        return circle;
+    },
+
+    triangleButton : () => {
+        let triangle = document.createElement('div');
+        triangle.setAttribute("class","triangle");
+        return triangle;
+    },
+
+    squareButton : () => {
+        let square = document.createElement('div');
+        square.setAttribute("class","square");
+        return square;
+    },
```
this approach again requires code change which can be made generic for minimal code change

ramkrishk commented on kids-learning-app/assets/js/app.js:
```diff
@@ -0,0 +1,260 @@
+        calculate.addEventListener("click",() => {
+            var inp = parseInt(inputBox.value);
+            if(inp <= 0 || inputBox.value === ''){
+                alert("Enter proper dimension.");
```
Good that the validation is taken care of, the user may not be aware of the value range, a detailed note would help.

Kamalesh-Radhakrishnan committed:
Pushed the common code to common selector.

Kamalesh-Radhakrishnan committed:
Alert Message detailed for user understanding.

Kamalesh-Radhakrishnan committed:
Review comments resolved

Added common DS for the object properties and behaviours. Removed hardcoded switch cases.
